modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationProcessor.java

```diff
@@ -0,0 +1,600 @@
+     * @param parallelismLevel Number of batches that can be handled simultaneously.
+     * @param batchSize Amount of keys to retrieve within one job.
+     * @param recheckAttempts Amount of potentially inconsistent keys recheck attempts.
+     * @param repairAlg Repair algorithm to be used to fix inconsistency.
+     * @param recheckDelay Specifies the time interval between two consequent attempts to check keys.
+     */
+    @SuppressWarnings("AssignmentOrReturnOfFieldWithMutableType")
+    public PartitionReconciliationProcessor(
+        long sesId,
+        IgniteEx ignite,
+        Collection<String> caches,
+        boolean fixMode,
+        int parallelismLevel,
+        int batchSize,
+        int recheckAttempts,
+        RepairAlgorithm repairAlg,
+        int recheckDelay
+    ) throws IgniteCheckedException {
+        super(sesId, ignite, parallelismLevel);
+
+        this.recheckDelay = recheckDelay;
+        this.caches = caches;
+        this.fixMode = fixMode;
+        this.batchSize = batchSize;
+        this.recheckAttempts = recheckAttempts;
+        this.repairAlg = repairAlg;
+    }
+
+    /**
+     * @return Partition reconciliation result
+     */
+    public ExecutionResult<PartitionReconciliationResult> execute() {
+        log.info(String.format(START_EXECUTION_MSG, fixMode, repairAlg, batchSize, recheckAttempts, parallelismLevel, caches));
+
+        try {
+            for (String cache : caches) {
+                IgniteInternalCache<Object, Object> cachex = ignite.cachex(cache);
+
+                Factory expiryPlcFactory = cachex.configuration().getExpiryPolicyFactory();
+                if (expiryPlcFactory != null && !(expiryPlcFactory.create() instanceof EternalExpiryPolicy)) {
+                    log.warning("The cache '" + cache + "' was skipped because CacheConfiguration#setExpiryPolicyFactory exists.");
+
+                    continue;
+                }
+
+                int[] partitions = ignite.affinity(cache).primaryPartitions(ignite.localNode());
+
+                for (int partId : partitions) {
+                    schedule(new Batch(sesId, UUID.randomUUID(), cache, partId, null));
+
+                    workProgress.assignWork();
+                }
+            }
+
+            boolean live = false;
+
+            while (!isEmpty() || (live = hasLiveHandlers())) {
+                if (topologyChanged())
+                    throw new IgniteException(TOPOLOGY_CHANGE_MSG);
+
+                if (isSessionExpired())
+                    throw new IgniteException(SESSION_CHANGE_MSG);
+
+                if (isInterrupted())
+                    throw new IgniteException(error.get());
+
+                if (isEmpty() && live) {
+                    Thread.sleep(100);
+
+                    continue;
+                }
+
+                PipelineWorkload workload = takeTask();
+
+                workProgress.printWorkProgress();
+
+                if (workload instanceof Batch)
```

Comment:
This if-else if-... could be moved to a separate method `handle(PipelineWorkload)` just for improving readability.